src/Payments.sol

```diff
@@ -0,0 +1,572 @@
+
+            settleRail(railIds[i], block.number);
+
+            Account storage account = accounts[rail.token][msg.sender];
+            account.lockupCurrent -= rail.lockupFixed + (rail.paymentRate * rail.lockupPeriod);
+            account.lockupRate -= rail.paymentRate;
+
+            rail.paymentRate = 0;
+            rail.lockupFixed = 0;
+            rail.lockupPeriod = 0;
+            rail.isActive = false;
+
+            OperatorApproval storage approval = operatorApprovals[rail.token][msg.sender][operator];
+            approval.rateAllowance = 0;
+            approval.lockupAllowance = 0;
+            approval.isApproved = false;
+        }
+    }
+
+    // TODO: implement
+    function terminateRail() public {
+
+    }
+
+    function deposit(address token, address to, uint256 amount) external {
+        require(token != address(0), "token address cannot be zero");
+        require(to != address(0), "to address cannot be zero");
+        require(amount > 0, "amount must be greater than 0");
+
+        // Create account if it doesn't exist
+        Account storage account = accounts[token][to];
+        if (account.ownerAddress == address(0)) {
+            account.ownerAddress = to;
+        }
+
+        // Transfer tokens from sender to contract
+        IERC20(token).safeTransferFrom(msg.sender, address(this), amount);
+
+        // Update account balance
+        account.funds += amount;
+
+        // settle account lockup now that we have more funds
+        settleAccountLockup(account);
+    }
+
+    function withdraw(address token, uint256 amount) external onlyAccountOwner(token) nonReentrant {
+        Account storage acct = accounts[token][msg.sender];
+
+        (bool funded, uint256 settleEpoch) = settleAccountLockup(acct);
+        require(funded && settleEpoch == block.number, "insufficient funds");
+
+        uint256 available = acct.funds > acct.lockupCurrent
+            ? acct.funds - acct.lockupCurrent
+            : 0;
+
+        require(amount <= available, "insufficient unlocked funds for withdrawal");
+        acct.funds -= amount;
+        IERC20(token).safeTransfer(msg.sender, amount);
+    }
+
+    function createRail(
+        address token,
+        address from,
+        address to,
+        address operator,
+        address arbiter
+    ) external returns (uint256) {
+        require(token != address(0), "token address cannot be zero");
+        require(from != address(0), "from address cannot be zero");
+        require(to != address(0), "to address cannot be zero");
+        require(operator != address(0), "operator address cannot be zero");
+
+        OperatorApproval memory approval = operatorApprovals[token][from][operator];
+        require(approval.isApproved, "operator not approved");
+
+        Account storage toAccount = accounts[token][to];
+        require(toAccount.ownerAddress != address(0), "to account does not exist");
+        require(toAccount.funds > 0, "to account has no funds");
```

Comment:
They don't need funds, do they?

Reply:
@Stebalien I mean not strictly speaking. We can get rid of this yeah.